primer/src/Primer/JSON.hs

```diff
@@ -43,4 +42,10 @@ import Deriving.Aeson.Stock (Vanilla)
 --
 -- * @SumTwoElemArray@ is unsupported by openapi3 as it is unrepresentable in
 --   a schema.
-type PrimerJSON a = Vanilla a
+type PrimerJSON a =
+  CustomJSON
+    '[ -- This protects some clients (e.g. TypeScript) from spurious equality failures
+       -- due to null-vs-omitted inconsistency.
+       OmitNothingFields
```

Comment:
In #966 etc, there was issues with `OmitNothingFields`. I assume that they are solved now that ghc has been upgraded? It may be good to keep this in mind in case there is a regression in the future